### ISSUE TITLE
Search order for boost-python library

### ISSUE THREAD

**Issue by ludwigschwardt** [casacore/python-casacore#167]
I find the following libraries installed by Homebrew:

boost-python:
```
/usr/local/Cellar/boost-python/1.69.0/lib/libboost_python27-mt.dylib
/usr/local/Cellar/boost-python/1.69.0/lib/libboost_python27.dylib
```
boost-python3:
```
/usr/local/Cellar/boost-python3/1.69.0/lib/libboost_python37-mt.dylib
/usr/local/Cellar/boost-python3/1.69.0/lib/libboost_python37.dylib
```

My casacore formula builds with thread support, I guess (`-DUSE_THREADS=ON`), so it links against the `mt` version of boost-python.

However, `python-casacore` looks up boost in the following way in `find_boost()`:
```
libboost_python-py37
libboost_python37
libboost_python
libboost_python3-mt
```

It then finds the non-`mt` version first and links against it. I'm guessing that's not a good idea...

Can we look for `mt` first or is that unsafe?

This could be related to #119 (I get the same error, although that issue claims that both are linked to the same boost-python).

**Comment by gijzelaerr** [comment]
This has been fixed right?

**Comment by ludwigschwardt** [comment]
Yes, fixed in #173. Forgot about this issue...
